scripts/deploy.sh

```diff
@@ -50,13 +50,32 @@ if [ "${DEPLOY_AIO}" != false ]; then
   #                  consume them.
   openstack-ansible -i 'localhost,' \
                     -e 'apt_target_group=localhost' \
+                    -e "apt_artifact_mode='${RPCO_APT_ARTIFACTS_MODE}'" \
                     -e 'container_artifact_enabled=false' \
                     -e 'py_artifact_enabled=false' \
                     "${SCRIPT_PATH}/../playbooks/site-artifacts.yml"
 
   # Install OpenStack-Ansible
   openstack-ansible "${SCRIPT_PATH}/../playbooks/openstack-ansible-install.yml"
 
+  # Set the AIO config bootstrap options
+  apt_artifact_enabled=$(awk '/apt_artifact_enabled/ {print $3}' /etc/ansible/facts.d/rpc_openstack.fact)
+  if [[ "${apt_artifact_enabled}" == "True" ]]; then
+    if [[ "${RPCO_APT_ARTIFACTS_MODE}" == "strict" ]]; then
+      # Prevent the AIO bootstrap from re-implementing
+      # the updates, backports and UCA sources. We have
+      # to use parameters here as BOOTSTRAP_OPTS only
+      # accepts string value parameters and the suffix
+      # parameter is a list.
+      echo -e '---\n{ "bootstrap_host_apt_distribution_suffix_list": [], "uca_enable": False }' > /opt/bootstrap-opts.yml
```

Comment:
Given this is something we're always going to do for the "strict" gate we should just carry this var file and point at it instead of creating the file within the script.

Reply:
I'll see whether this could be re-implemented through the use of local facts or something, given that the AIO bootstrap does not use any of the /etc/openstack_deploy files (nor should it). I'll need to fire up a test env to verify so I'll pick it up tomorrow.